source/lac/scalapack.cc

```diff
@@ -667,26 +669,85 @@ void ScaLAPACKMatrix<NumberType>::compute_cholesky_factorization()
       ppotrf(&uplo,&n_columns,A_loc,&submatrix_row,&submatrix_column,descriptor,&info);
       AssertThrow (info==0, LAPACKSupport::ExcErrorCode("ppotrf", info));
     }
-  property = (uplo=='L' ? LAPACKSupport::lower_triangular : LAPACKSupport::upper_triangular);
   state = LAPACKSupport::cholesky;
+  property = (uplo=='L' ? LAPACKSupport::lower_triangular : LAPACKSupport::upper_triangular);
 }
 
 
 
 template <typename NumberType>
-void ScaLAPACKMatrix<NumberType>::invert()
+void ScaLAPACKMatrix<NumberType>::compute_lu_factorization()
 {
-  if (state == LAPACKSupport::matrix)
-    compute_cholesky_factorization();
+  Assert (state == LAPACKSupport::matrix,
+          ExcMessage("Matrix has to be in Matrix state before calling this function."));
 
   if (grid->mpi_process_is_active)
     {
       int info = 0;
       NumberType *A_loc = &this->values[0];
-      ppotri (&uplo,&n_columns, A_loc, &submatrix_row, &submatrix_column, descriptor,&info);
-      AssertThrow (info==0, LAPACKSupport::ExcErrorCode("ppotri", info));
+
+      const int iarow = indxg2p_(&submatrix_row, &row_block_size, &(grid->this_process_row), &first_process_row, &(grid->n_process_rows));
+      const int mp = numroc_(&n_rows, &row_block_size, &(grid->this_process_row), &iarow, &(grid->n_process_rows));
+      ipiv.resize(mp+row_block_size);
+
+      pgetrf(&n_rows,&n_columns,A_loc,&submatrix_row,&submatrix_column,descriptor,ipiv.data(),&info);
+      AssertThrow (info==0, LAPACKSupport::ExcErrorCode("pgetrf", info));
+    }
+  state = LAPACKSupport::State::lu;
+  property = LAPACKSupport::Property::general;
+}
+
+
+
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::invert()
+{
+  // Check whether matrix is symmetric and save flag.
+  // If a Cholesky factorization has been applied previously,
+  // the original matrix was symmetric.
+  bool is_symmetric = (property == LAPACKSupport::symmetric || state == LAPACKSupport::State::cholesky) ? true : false;
```

Comment:
you don't need the `?` part, also make `const`, i.e.
```
const bool is_symmetric = (property == LAPACKSupport::symmetric || state == LAPACKSupport::State::cholesky);
```

Reply:
Done!